src/Http/Cookie/Cookie.php

```diff
@@ -466,6 +484,51 @@ public function withExpired()
         return $new;
     }
 
+    /**
+     * Get the SameSite attribute.
+     *
+     * @return string|null
+     */
+    public function getSameSite()
+    {
+        return $this->sameSite;
+    }
+
+    /**
+     * Create a cookie with an updated SameSite option.
+     *
+     * @param string|null $sameSite Value for to set for Samesite option.
+     *   One of CookieInterface::SAMESITE_* constants.
+     * @return static
+     */
+    public function withSameSite($sameSite = null)
+    {
+        if ($sameSite !== null) {
+            $this->validateSameSiteValue($sameSite);
+        }
+
+        $new = clone $this;
+        $new->sameSite = $sameSite;
+
+        return $new;
+    }
+
+    /**
+     * Check that value passed for SameSite is valid.
+     *
+     * @param string $sameSite SameSite value
+     * @return void
+     * @throws \InvalidArgumentException
+     */
+    protected static function validateSameSiteValue($sameSite)
+    {
+        if (!in_array($sameSite, CookieInterface::SAMESITE_VALUES, true)) {
+            throw new InvalidArgumentException(
+                'Samesite value must be either of: ' . implode(', ', CookieInterface::SAMESITE_VALUES)
```

Comment:
```suggestion
                'SameSite value must be either of: ' . implode(', ', CookieInterface::SAMESITE_VALUES)
```
Should this be the same spelling as constructor parameter, attribute name or normal english?

Reply:
if it is the key, I would code fence it: `samesite` etc.